### ISSUE TITLE
Eliminate 'ThemeChanged' warning when running IDLE tests

### ISSUE THREAD

**Issue by terryjreedy** [python/cpython#71383]
BPO | [27196](https://bugs.python.org/issue27196)
--- | :---
Nosy | @terryjreedy, @ned-deily, @serhiy-storchaka

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/terryjreedy'
closed_at = None
created_at = <Date 2016-06-03.08:00:06.971>
labels = ['expert-IDLE', 'type-bug', 'expert-tkinter', '3.9']
title = "Eliminate 'ThemeChanged' warning when running IDLE tests"
updated_at = <Date 2020-01-07.22:42:55.465>
user = 'https://github.com/terryjreedy'
```

bugs.python.org fields:
```python
activity = <Date 2020-01-07.22:42:55.465>
actor = 'terry.reedy'
assignee = 'terry.reedy'
closed = False
closed_date = None
closer = None
components = ['IDLE', 'Tkinter']
creation = <Date 2016-06-03.08:00:06.971>
creator = 'terry.reedy'
dependencies = []
files = []
hgrepos = []
issue_num = 27196
keywords = []
message_count = 7.0
messages = ['267078', '267081', '267226', '267228', '268861', '269089', '359559']
nosy_count = 4.0
nosy_names = ['terry.reedy', 'ned.deily', 'python-dev', 'serhiy.storchaka']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'needs patch'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue27196'
versions = ['Python 3.9']
```

</p></details>

### Linked PRs
* gh-104463
* gh-104464
* gh-108806
* gh-112973

**Comment by terryjreedy** [comment]
The 2010-date tcl-tk ticket says that this is known when "an interpreter is created and then immediately destroyed".  Last reproduced in Sept 2022 with 8.6 on Win 10.  I believe I have seen without immediate close, but will see if possible.

I currently see 3 with `python -m test.test_idle`.  Adding `-v` and the messages appear after
test_iomenu.IOBindingTest.test_init
test_searchbase.SearchDialogBaseTest.test_open_and_close
test_textview.ButtonClickTest.test_view_text_bind_with_button
These are each the last test in the file.

The above is not directly useful as the current glitches disappear when running 1 test file.  I discovered that the first appears when test_history, test_hyperparser, and test_iomenu are run together, but not when iomenu or the other two are run separately.  My next PR will be to update the 2 master test files and add a comment in idle_test.__init__ on how to run subsets.

**Comment by chrstphrchvz** [comment]
Tcl/Tk has since merged a fix which will appear in 8.6.14. However, a patch for the fix is small and should apply cleanly to the past few Tk releases. Is Python opposed to backporting fixes for its bundled Tcl/Tk?

**Comment by terryjreedy** [comment]
When would you expect 8.6.14?

**Comment by chrstphrchvz** [comment]
The past 6 Tcl/Tk 8.6.x releases were around November each year, when the release manager got around to doing so and after any critical fixes were made to Tk Aqua for the latest macOS. So I would guess 8.6.14 is still several months away.

**Comment by chrstphrchvz** [comment]
I have opened #108806 and python/cpython-source-deps#29 in case Python is interested in applying the upstream fix to the bundled Tcl/Tk 8.6.8 and 8.6.13 in the macOS and Windows installers.

**Comment by ned-deily** [comment]
Thanks for the PRs. The fix for Tk in the macOS installer has now been merged for release in 3.13.0a2, 3.12.1, and 3.11.7.

**Comment by zooba** [comment]
Windows fix is now committed for 3.13. As discussed elsewhere, I'm not comfortable backporting the patched build to 3.12 or earlier without someone present to promptly debug and fix any issues that arise.

**Comment by terryjreedy** [comment]
Unless Serhiy wants to do the Windows backports, I am happy to have this closed.  I currently cannot repeat the problems on 3.11/12 Windows I reported last May in https://github.com/python/cpython/issues/71383#issuecomment-1546792789.  (This is not the first time the messages have appeared and disappeared.)  And the patch should prevent any future appearances.  If something reappears in 3.12, I now know not to waste more time trying to make it go away.  If something reappears in main after this patch, it should be a new issue.
